package.json

```diff
@@ -115,5 +115,6 @@
   },
   "tsd": {
     "directory": "test/types"
-  }
+  },
+  "packageManager": "yarn@1.22.19"
```

Comment:
please don't add this

Reply:
this line is necessary, or else installation fails on computers with any modern version of yarn installed

Reply:
Feel free to add it locally, but don't commit it to the repository. We are not using yarn

Reply:
more specifically, run `corepack enable`, and then clone this repo and run `yarn install` and then watch what happens

Reply:
oh, nevermind it's been too long and im remembering now that this repo doesn't prescribe a package manager for some reason

Reply:
that's way beyond the scope of this PR.
to my knowledge, none of fastify or pino dependencies have this field set. If there are good reasons to set it - please open an issue describing the problem and giving some references about this field.

Reply:
sorry, i got confused because this repo doesn't use a package manager, so installation defaulted to yarn, which caused the issue. as an aside, why isn't there a lock file committed to git? that's non-idiomatic afaict

Reply:
it's a library, not an application. package-lock is not applied transitively

Reply:
From your message, you seem to be implying that it is not correct for JavaScript/TypeScript libraries to commit a package lock file. But is that true? [Sindre Sorhus seems to echo that opinion here](https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514). He specifically gives the justification of "Lockfiles are great for apps where you want a controlled reproducible environment, but for packages this doesn't make much sense."

But consider this very issue with the `tsd` dependency - we very much want exact reproducible builds when troubleshooting CI issues for our libraries! Furthermore, as far as I can tell, the ecosystem disagrees with this, and it is idiomatic to commit a lockfile for a library. For example:

- [TypeScript has a package-lock.json file](https://github.com/microsoft/TypeScript/blob/main/package-lock.json)
- [React has a yarn.lock file](https://github.com/facebook/react/blob/main/yarn.lock)
- [Angular has a yarn.lock file](https://github.com/angular/angular/blob/main/yarn.lock)
- [Redux has a yarn.lock file](https://github.com/reduxjs/redux/blob/master/yarn.lock)
- [Next.js has a pnpm-lock.yaml file](https://github.com/vercel/next.js/blob/canary/pnpm-lock.yaml)
- [Vue has a pnpm-lock.yaml file](https://github.com/vuejs/core/blob/main/pnpm-lock.yaml)
- [Moment.js has a package-lock.json file](https://github.com/moment/moment/blob/develop/package-lock.json)
- [JQuery has a package-lock.json file](https://github.com/jquery/jquery/blob/main/package-lock.json)
- [Nest.js has a package-lock.json file](https://github.com/nestjs/nest/blob/master/package-lock.json)
- [Express does not have a lock file](https://github.com/expressjs/express)
- [Astro has a pnpm-lock.yaml file](https://github.com/withastro/astro/blob/main/pnpm-lock.yaml)
- [Quik has a pnpm-lock.yaml file](https://github.com/BuilderIO/qwik/blob/main/pnpm-lock.yaml)
- [Remix has a yarn.lock file](https://github.com/remix-run/remix/blob/main/yarn.lock)
- [Axios has a package-lock.json file](https://github.com/axios/axios/blob/v1.x/package-lock.json)

But you mostly answered the original question, so I'm satisfied for now, thanks.